testapi.pm

```diff
@@ -59,13 +61,29 @@ sub record_soft_failure {
 sub assert_screen($;$) {
     my ($mustmatch, $timeout) = @_;
     bmwqemu::log_call('assert_screen', mustmatch => $mustmatch, timeout => $timeout);
-    return bmwqemu::assert_screen(mustmatch => $mustmatch, timeout => $timeout);
+    return $last_matched_needle = bmwqemu::assert_screen(mustmatch => $mustmatch, timeout => $timeout);
 }
 
 sub check_screen($;$) {
     my ($mustmatch, $timeout) = @_;
     bmwqemu::log_call('check_screen', mustmatch => $mustmatch, timeout => $timeout);
-    return bmwqemu::assert_screen(mustmatch => $mustmatch, timeout => $timeout, check => 1);
+    return $last_matched_needle = bmwqemu::assert_screen(mustmatch => $mustmatch, timeout => $timeout, check => 1);
+}
+
+sub match_has_tag($) {
+    my ($tag) = @_;
+    if ($last_matched_needle) {
+        return $last_matched_needle->{needle}->has_tag($tag);
+    }
+    return undef;
+}
+
+sub match_is_workaround($) {
+    my ($tag) = @_;
```

Comment:
why not just hardcoded $tag = 'workaround'; in case the function named match_is_workaround?